internal/pkg/checks/check.go

```diff
@@ -40,15 +42,28 @@ type Check struct {
 }
 
 type CapturedRequest struct {
-	StatusCode int
-	TestId     string
-	Path       string
-	Host       string
+	TestId  string
+	Path    string
+	Host    string
+	Method  string
+	Proto   string
+	Headers map[string][]string
 }
 
-func captureRequest(location string, hostOverride string) (data CapturedRequest, err error) {
+type CapturedResponse struct {
+	StatusCode    int
+	ContentLength int64
+	Proto         string
+	Headers       map[string][]string
+}
```

Comment:
I'm not sure I see the benefits, but I'd be happy to track this request in a github issue.

Reply:
Yeh, this is easily a matter of taste. I thought about this approach because I was thinking about ways to avoid using names return in `captureRequest`. I can prototype a PR to see what you think.